src/main/kotlin/com/exactpro/th2/check1/rule/CheckTaskUtils.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+@file:JvmName("CheckTaskUtils")
+package com.exactpro.th2.check1.rule
+
+import com.exactpro.th2.check1.AbstractSessionObserver
+import com.exactpro.th2.common.event.Event
+import com.exactpro.th2.common.event.EventUtils
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.grpc.EventBatch.Builder
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.EventStatus
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.protobuf.ByteString
+import com.google.protobuf.TextFormat
+import java.lang.IllegalArgumentException
+import com.exactpro.th2.common.grpc.Event as ProtoEvent
+
+val OBJECT_MAPPER = ObjectMapper()
+
+fun Event.disperseToBatches(maxEventBatchContentSize: Int, parentEventId: EventID): List<EventBatch> {
+    if(maxEventBatchContentSize <= 0) {
+        throw IllegalArgumentException("'maxEventBatchContentSize' should be greater than zero, actual: $maxEventBatchContentSize")
+    }
+
+    val events = toListProto(parentEventId)
+
+    val result = ArrayList<EventBatch>()
+    val eventGroups = events.groupBy(ProtoEvent::getParentId)
+    batch(maxEventBatchContentSize, result, eventGroups, parentEventId)
+    return result
+}
+
+private fun batch(maxEventBatchContentSize: Int, result: MutableList<EventBatch>, eventGroups: Map<EventID, List<com.exactpro.th2.common.grpc.Event>>, eventID: EventID) {
```

Comment:
You can add an alias for `com.exactpro.th2.common.grpc.Event` in import to avoid writing the fully qualified name